File: mobile/android/app/src/main/java/com/korzadi/vpn/wireguard/WireGuardManager.java
package com.korzadi.vpn.wireguard;


public class WireGuardManager {


    private boolean connected = false;


    public boolean connect(
            WireGuardProfile profile
    ){

        connected = true;

        return connected;
    }


    public boolean disconnect(){

        connected = false;

        return true;
    }


    public boolean isConnected(){

        return connected;
    }
}

File: mobile_kmp_restore_candidate/shared/src/desktopMain/kotlin/com/korzadi/vpn/vpn/WireGuardManager.kt
package com.korzadi.vpn.vpn

import kotlinx.coroutines.flow.MutableStateFlow

actual class WireGuardManager actual constructor() {
    private val _state = MutableStateFlow(VPNState.DISCONNECTED)
    
    actual fun connect(config: VPNConfig) {
        _state.value = VPNState.CONNECTING
        // Lógica de wg-quick
        _state.value = VPNState.CONNECTED
    }

    actual fun disconnect() {
        _state.value = VPNState.DISCONNECTING
        // Lógica de desconexión
        _state.value = VPNState.DISCONNECTED
    }

    actual fun status(): VPNState {
        return _state.value
    }
}
